@utahdts/utah-design-system/react/components/containers/tabs/TabGroup.jsx

```diff
@@ -38,12 +50,39 @@ export function TabGroup({
           setTabGroupState((draftState) => { draftState.selectedTabId = tabId; });
         }
       },
+      navigateNext: () => {
+        const index = findCurrentTabIndex();
+        if (index !== (tabGroupState.tabs.length - 1)) {
+          navigateTab(tabGroupState?.tabs?.[index + 1] || null);
+        } else {
+          navigateTab(tabGroupState?.tabs?.[0] || null);
+        }
+      },
+      navigatePrevious: () => {
+        const index = findCurrentTabIndex();
+        if (index !== 0) {
+          navigateTab(tabGroupState?.tabs?.[index - 1] || null);
+        } else {
+          navigateTab(tabGroupState?.tabs?.[tabGroupState.tabs.length - 1] || null);
+        }
+      },
+      isVertical: !!isVertical,
     }),
     [onChange, setTabGroupState, tabGroupState, value]
   );
+
+  useEffect(() => {
+    if (tabGroupRef?.current) {
+      setTabGroupState((draftState) => {
+        // @ts-ignore
+        draftState.tabs = [...tabGroupRef?.current?.querySelectorAll('button[role=tab]') || []];
```

Comment:
Other places use `register` and `unregister` methods in the state for children to register that they exist instead of querying the javascript directly. You can see an example of this in ComboBoxOption where it calls `registerOption` and `unregisterOption`.

There's also the possibility of storing the current tab index in state instead of all the tabs?

If a tab were added dynamically, this would not pick it up since it only fires once after the first render. Also, would there ever be a tab group inside of a tab group 😲. If so, this JS query would over fetch.